spyder/config/manager.py

```diff
@@ -52,6 +56,33 @@ def __init__(self, parent=None, active_project_callback=None):
         # Setup
         self.remove_deprecated_config_locations()
 
+    def register_plugin(self, plugin_class):
+        """Register plugin configuration."""
+        conf_section = plugin_class.CONF_SECTION
+        if plugin_class.CONF_FILE and conf_section:
+            path = self.get_plugin_config_path(conf_section)
+            version = plugin_class.CONF_VERSION
+            version = version if version else '0.0.0'
+            name_map = plugin_class.CONF_NAME_MAP
+            name_map = name_map if name_map else {'spyder': []}
+            defaults = plugin_class.CONF_DEFAULTS
+
+            if conf_section in self._plugin_configs:
+                raise Exception('A plugin with section "{}" already exsists!'

```

Comment:
exsists -> exists

Reply:
Also, perhaps it'd be better to use `RuntimeError` here instead of `Exception`, which is too broad.